neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -221,6 +221,7 @@ type Guest struct {
 	MemorySlotSize resource.Quantity `json:"memorySlotSize"`
 	// +optional
 	MemorySlots MemorySlots `json:"memorySlots"`
+	// Deprecated: MemoryProvider is ignored and always interpreted as equal to VirtioMem.
```

Comment:
are we going to drop the variable eventually?

Reply:
Yeah -- the idea is that this PR i just the first half, making making this field not have an effect (but still set it to to VirtioMem). The second half will be to remove the field.

(We need to do it in two steps so that each one is safe to roll back)